apps/dolly-backend/src/main/java/no/nav/dolly/service/excel/BankkontoExcelService.java

```diff
@@ -121,28 +186,58 @@ public Mono<Void> prepareBankkontoSheet(XSSFWorkbook workbook, List<String> test
         return Mono.empty();
     }
 
-    private List<Object[]> getBankkontoDetaljer(List<String> identer) {
+    private Mono<List<Object[]>> tpsBankkonto(List<String> identer) {
+        return Flux.range(0, identer.size())
+                .flatMap(index -> tpsMessagingConsumer.getPersoner(List.of(identer.get(index)), List.of("q1")))
+                .filter(PersonMiljoeDTO::isOk)
+                .map(PersonMiljoeDTO::getPerson)
+                .map(person -> unpackBankkonto(person))
+                .collectList();
+    }
 
-        var bankKontoIdenter = identRepository.findByIdentIn(identer)
-                .stream()
-                .map(Testident::getTestgruppe)
-                .map(Testgruppe::getBestillinger)
-                .flatMap(Collection::stream)
+    private Mono<List<Object[]>> kontoregisterBankkonto(List<String> identer) {
+        return Flux.range(0, identer.size())
+                .flatMap(index -> kontoregisterConsumer.sendHentKontoRequest(identer.get(index)))
+                .filter(konto -> konto != null && konto.getAktivKonto() != null)
+                .map(konto -> unpackBankkonto(konto.getAktivKonto()))
+                .collectList();
+    }
+
+    private List<Object[]> getBankkontoDetaljer(Testgruppe testgruppe) {
+
+        var bankKontoIdenter = testgruppe.getBestillinger().stream()
                 .filter(bestilling -> nonNull(bestilling.getBestKriterier()))
                 .filter(bestilling -> bestilling.getBestKriterier().contains("Bankkonto"))
                 .map(Bestilling::getProgresser)
                 .flatMap(Collection::stream)
-                .map(BestillingProgress::getIdent)
-                .distinct()
+                .collect(Collectors.teeing(
+                        Collectors.filtering(
+                                p -> !p.getBestilling().getSistOppdatert().isAfter(NY_KONTOREGISTER_I_BRUK),
+                                Collectors.toList()
+                        ),
+                        Collectors.filtering(
+                                p -> p.getBestilling().getSistOppdatert().isAfter(NY_KONTOREGISTER_I_BRUK)
+                                        && !p.getKontoregisterStatus().contains("Feil"),
+                                Collectors.toList()
+                        ),
+                        List::of
```

Comment:
Kunne dette blitt tydligere med en map av lister, som her: 
https://www.educative.io/answers/what-is-collectorsteeing-in-java

Reply:
ja, da er det mer klar hva den list er, fikser.